### DISCUSSION TITLE
Can't access backend after Android build, but works on webpage

### DISCUSSION THREAD

**Discussion by athelene** [quasarframework/quasar#13532] [CLI - Capacitor / Cordova modes]
I am very new to Quasar. I've built an app that builds and works as a web app. When I build it as an Android app, the front-end comes up in Android Studio. The first page is a login page. When I log in with valid credentials, I'm getting a 404 error. Yet the backend responds perfectly for the web app both in dev and production. On the backend, it looks like the app isn't even reaching the server, there's nothing logging there (as it does from the web app.) 

Originally, I had a CORS error. To avoid CORS issues, I added a header to the backend to accept the external IP address that is used in Android Studio. Now that error is gone but I'm seeing this specific 404 error: 

Line 311 - Msg: Uncaught (in promise) Error: Request failed with status code 404

Hoping someone can help me figure out what troubleshooting steps I can take. Not sure why the call to the backend would be different between the two.

**Comment by metalsadman** [comment]
check in your logs what URL is actually being called.

**Reply by athelene** [reply]
Thanks for the response. Now I can't even find where I saw the 404 error. 

I'm not sure how to use the Android Studio tools (or something else?) to see what I normally see in the terminal when I'm troubleshooting a web app. Yesterday, when I posted the question I think I was in the debugger and console area. But this morning all I'm seeing is  Msg: Uncaught (in promise) Error: Network Error.  I thought I was able to get into a network section somewhere where I saw the 404. But I can't see it now. At this point, I'd take a link to a good tutorial on what tools to use for troubleshooting Android apps and how to use them.  Also, something that would explain how an app appears to an API when using a tool like this. In other words, how should I set up the header in the api so that it will be accepted when accessed via a mobile app? What address should it be expecting and allowing? I can't imagine that you'd want to open it up to everything? That said, it really feels like it's not even reaching my api at all since I'm not seeing the expected logging I've set up when the api is used. 

I realize this question is not specific. Unfortunately, I can't even figure out where to start to get the information I need. Any help or direction is much appreciated.

**Comment by athelene** [comment]
On my web app, I have been using my api server in dev along with my vue/quasar web app in dev. Then, when I build for production, I use the production version of both. To try to get past my problem, I set up my dev environment to use the production api server. Then I restarted quasar/android studio and now my android app is working in the emulator. I'd prefer not to use anything production related while developing, but this will allow me to keep moving forward. Maybe this will help someone down the road who might face a similar problem. At this point I'm guessing that my problem has something to do with the fact that I was accessing an api on localhost. I'll close out the question since it's pretty vague and I'm now up and running.  Thanks again for taking a minute to respond.
